components/switch/src/switch.js

```diff
@@ -0,0 +1,165 @@
+/** @jsx jsx */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+import { SwitchText, SwitchToggle } from './styled';
+
+// ==============================
+// Utils
+// ==============================
+
+// ==============================
+// Component
+// ==============================
+
+export const Switch = ({
+	name,
+	id,
+	toggleText,
+	size,
+	block,
+	flip,
+	srOnlyText,
+	defaultChecked,
+	disabled,
+	children,
+	...props
+}) => {
+	const { switch: formSwitch } = useTheme();
+
+	const [isChecked, setIsChecked] = useState(defaultChecked);
+	const toggle = () => setIsChecked(!isChecked);
+
+	// Common styling
+	const common = {
+		position: 'relative',
+		display: block ? 'flex' : 'inline-flex',
+		flexWrap: 'wrap',
+		verticalAlign: 'middle',
+		marginRight: block ? null : formSwitch.marginRight,
+		marginBottom: formSwitch.marginBottom,
+		alignItems: 'center',
+		width: block ? '100%' : null,
+		flexDirection: flip ? 'row-reverse' : null,
+
+		':hover': {
+			cursor: 'pointer',
+		},
+	};
+
+	return (
+		<label htmlFor={id} css={common} onChange={toggle} {...props}>
+			<input
+				type="checkbox"
+				css={{
+					position: 'absolute',
+					zIndex: -1,
+					opacity: 0,
+				}}
+				name={name}
+				id={id}
```

Comment:
We should auto-gen this id. Is there ever a case for them to pass in their own id? Would ensure we are always accessible

Reply:
Not sure

Is it best practise to auto-generate ID values within the component?

Can we mark this for discussion with TM?

Reply:
Found this... 

https://codeburst.io/unique-ids-and-where-to-find-them-14020ddbc964

Reply:
Yea we can mark as discussion. I like autogenerating as it ensures that there is always a unique id and makes one less thing for the dev to worry about.

Example in modal
https://github.com/WestpacGEL/GEL/blob/689c5a5061fc24d0d33287a76f7ab246714dcbdb/components/modal/src/Modal.js#L35

Reply:
I've implemented auto-generate as above :)